### PR TITLE
Adding a warning when loading a config YAML with a docker_images section, instead of an ad_identifiers section

### PR DESCRIPTION
**Pull request by jiquiame**
### What does this PR do?

Forbid the load of a YAML with a `docker_images:` section so that check configs copied from version 5 of the agent are not loaded as static checks. Add a Warning to explain this.

### Motivation

This could be confusing for troubleshooting.

### CODE REVIEW

File: pkg/collector/providers/file.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ type configFormat struct {
 	MetricConfig  interface{} `yaml:"jmx_metrics"`
 	LogsConfig    interface{} `yaml:"logs"`
 	Instances     []check.ConfigRawMap
+	DockerImages []string `yaml:"docker_images"` // To avoid issues if check uses former conf
```
Comment (xvello) [inline pkg/collector/providers/file.go]: Let's change the comment to `// Only imported for deprecation warning`

File: pkg/collector/providers/file.go [hunk 2]
```diff
@@ -295,5 +296,10 @@ func GetCheckConfigFromFile(name, fpath string) (check.Config, error) {
 		config.LogsConfig = rawLogsConfig
 	}
 
+	// DockerImages entry was found: we ignore it if no ADIdentifiers has been found
+	if cf.DockerImages != nil && cf.ADIdentifiers == nil {
```
Comment (xvello) [inline pkg/collector/providers/file.go]: Although `[]string{nil}` is a thing, using that might break if the yaml lib changes it's behaviour to serialise in empty `[]string{}`s instead. WDYT about checking the length instead?

File: pkg/collector/providers/file.go [hunk 3]
```diff
@@ -295,5 +296,10 @@ func GetCheckConfigFromFile(name, fpath string) (check.Config, error) {
...
+		return config, errors.New("the 'docker_images:' section is deprecated, please use 'ad_identifiers:' instead")
```
Comment (xvello) [inline pkg/collector/providers/file.go]: I'd remove the `:` at the end of the section names for readability

**Review by xvello — CHANGES_REQUESTED** [review]
Some nits on the code. Also, could you please add a `ad-deprecated.yaml` fixture and a test case to `TestGetCheckConfig`?

File: pkg/collector/providers/file.go (hunk 3 above)
Comment (antoinepouille) [reply pkg/collector/providers/file.go]: 👍

File: pkg/collector/providers/file.go (hunk 1 above)
Comment (antoinepouille) [reply pkg/collector/providers/file.go]: Much better indeed

**Review by xvello — APPROVED** [review]
LGTM

**Pull request merged**